src/Pos/Launcher/Resource.hs

```diff
@@ -0,0 +1,380 @@
+import           Pos.Slotting                 (SlottingContextSum (..), SlottingVar,
+                                               mkNtpSlottingVar, runSlotsDataRedirect,
+                                               runSlotsRedirect)
+import           Pos.Ssc.Class                (SscConstraint, SscParams,
+                                               sscCreateNodeContext)
+import           Pos.Ssc.Extra                (SscState, mkSscState)
+import           Pos.Txp                      (GenericTxpLocalData, mkTxpLocalData)
+#ifdef WITH_EXPLORER
+import           Pos.Explorer                 (explorerTxpGlobalSettings)
+#else
+import           Pos.Txp                      (txpGlobalSettings)
+#endif
+import           Pos.Security                 (SecurityWorkersClass)
+import           Pos.Update.Context           (UpdateContext (..))
+import qualified Pos.Update.DB                as GState
+import           Pos.Update.MemState          (newMemVar)
+import           Pos.Util.Concurrent.RWVar    as RWV
+import           Pos.Worker                   (allWorkersCount)
+import           Pos.WorkMode                 (TxpExtra_TMP)
+
+-- Remove this once there's no #ifdef-ed Pos.Txp import
+{-# ANN module ("HLint: ignore Use fewer imports" :: Text) #-}
+
+----------------------------------------------------------------------------
+-- Data type
+----------------------------------------------------------------------------
+
+-- | This data type contains all resources used by node.
+data NodeResources ssc m = NodeResources
+    { nrContext    :: !(NodeContext ssc)
+    , nrDBs        :: !NodeDBs
+    , nrSscState   :: !(SscState ssc)
+    , nrTxpState   :: !(GenericTxpLocalData TxpExtra_TMP)
+    , nrDlgState   :: !DelegationVar
+    , nrPeerState  :: !(PeerStateCtx Production)
+    , nrTransport  :: !(Transport m)
+    , nrJLogHandle :: !(Maybe Handle)
+    -- ^ Handle for JSON logging (optional).
+    }
+
+hoistNodeResources ::
+       forall ssc n m. Functor m
+    => (forall a. n a -> m a)
+    -> NodeResources ssc n
+    -> NodeResources ssc m
+hoistNodeResources nat nr =
+    nr {nrTransport = hoistTransport nat (nrTransport nr)}
+
+----------------------------------------------------------------------------
+-- Allocation/release/bracket
+----------------------------------------------------------------------------
+
+-- | Allocate all resources used by node. They must be released eventually.
+allocateNodeResources ::
+       forall ssc m.
+       ( SscConstraint ssc
+       , SecurityWorkersClass ssc
+       , WithLogger m
+       , MonadIO m
+       , MonadMockable m
+       , MonadMask m
+       , MonadBase IO m
+       , MonadBaseControl IO m
+       )
+    => NodeParams
+    -> SscParams ssc
+    -> m $ NodeResources ssc m
+allocateNodeResources np@NodeParams {..} sscnp = do
+    db <- openNodeDBs npRebuildDb npDbPathM
+    -- TODO: usage of 'runResourceT' here is strange.
```

Comment:
At least because there is no resource when I want to do a simple `get`. Anyway, this line was removed during the merge.